webhook_to_fedora_messaging/endpoints/parser/parser.py

```diff
@@ -0,0 +1,53 @@
+from  flask import Request, request
+from werkzeug.datastructures.headers import EnvironHeaders
+import hashlib
+import hmac
+from webhook_to_fedora_messaging_messages.github.github import GithubMessageV1
+from ..util import fas_by_github
+from ...exceptions.exceptions import SignatureMatchError, SecretNotFoundError
+
+
+def msg_parser(service_type: str, secret: str):
+    if service_type.lower() == "github":
+        return _github_parser(secret)
+    else:
+        raise ValueError
+
+
+def _github_parser(secret: str) -> GithubMessageV1:
+    """Convert Flask request objects into desired FedMsg format.
+    
+    Args:
+        headers: Flask request.headers
+        body: A dictionary contains payload
+        secret: Specifies whether the webhook has secret key feature on or not
+    """
+    
+    headers = dict(request.headers)
+    
+    if secret and not verify_signature(secret, headers['X-Hub-Signature-256']):
+        return SignatureMatchError("Message Signature Couldn't be Matched.")
+
+    topic = "github." + headers['X-Github-Event']
```

Comment:
It's preferable to use string formatting rather than concatenation, for performance reasons:

```suggestion
    topic = f"github.{headers['X-Github-Event']}"
```